pkg/node/address.go

```diff
@@ -441,12 +441,8 @@ func chooseHostIPsToRestore(ipv6 bool, fromK8s, fromFS net.IP, cidr *cidr.CIDR)
 			// case that the IP from the FS is not within the CIDR. If we
 			// fallback, then we also need to check the fromK8s IP is also
 			// within the CIDR.
-			if cidr != nil && cidr.Contains(ip) {
-				return
-			} else if cidr != nil && cidr.Contains(fromK8s) {
-				ip = fromK8s
-				return
-			}
+			ip = fromK8s
```

Comment:
The comment above says IP from FS is preferred over the one from k8s. Let's retain the same behavior ?